mlrun/model_monitoring/api.py

```diff
@@ -561,9 +561,10 @@ def _create_model_monitoring_function_base(
             "An application cannot have the following names: "
             f"{mm_constants._RESERVED_FUNCTION_NAMES}"
         )
-    if name and name.endswith(mm_constants._RESERVED_EVALUATE_FUNCTION_SUFFIX):
+    _, was_changed, suffix = mlrun.utils.helpers.ensure_batch_job_suffix(name)
+    if name and not was_changed:
```

Comment:
I find the new condition harder to understand. How about defining `ends_with_batch = not was_changed`?